presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeIcebergTpcdsWithPositionalDeletes.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.Session;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.testng.Assert.assertEquals;
+
+@Test(groups = {"parquet"})
+public class TestPrestoNativeIcebergTpcdsWithPositionalDeletes
+        extends TestPrestoNativeIcebergTpcdsQueriesParquetUsingThrift
+{
+    Map<String, Long> deletedRowsMap = new HashMap<>();
+
+    @Override
+    protected void doDeletes()
```

Comment:
These delete queries are the same for positional delete and equality delete, and . I think they better be in the parent class `TestPrestoNativeIcebergTpcdsQueriesParquetUsingThrift`.

Reply:
Presto doesn't support writing equality delete yet, so it's ok to remove this class after moving all common queries to the parent class. In the future if equality delete yet is supported, then we can subclass TestPrestoNativeIcebergTpcdsQueriesParquetUsingThrift to TestPrestoNativeIcebergTpcdsQueriesParquetUsingThriftWithPositionalDelete and TestPrestoNativeIcebergTpcdsQueriesParquetUsingThriftWithEqualityDelete, and these two classes should be only a few lines which is used to set the delete method.

Reply:
We are doing the equality deletes using the Iceberg API, since there is no support to enforce equality deletes using DELETE. So this function will have a different implementation for equality delete test case.

Reply:
I will move it now to base class. If there is support to enforce the type of delete in the future, we can keep it here. Or else, we will move it back to derived classes.